### PR TITLE
Fix get_parent_range to return 0 when there is no parent

### LINKED ISSUE #15635: Wrong Control size for full rect elements when margins are 0
**Godot version:** current master, introduced by #15610 @groud 

**Issue description:**

Wrong Control size for full rect elements when margins are 0

**Steps to reproduce:**

Just create a control, set full rect -> save scene -> close scene -> reopen -> the margin is now read as -1.

### PR DESCRIPTION
**Pull request by endragor**
Fixes #15635

### CODE REVIEW

**Comment by endragor** [comment]
0 makes more sense here and doesn't seem to cause problems wherever _get_parent_range is used. I wonder why 1.0 was there before... @reduz ?

File: scene/gui/control.cpp [hunk 1]
```diff
@@ -1334,7 +1334,7 @@ float Control::_get_parent_range(int p_idx) const {
 		return get_viewport()->get_visible_rect().size[p_idx & 1];
 	}
 
-	return 1.0;
+	return 0;
```
Comment (groud) [inline scene/gui/control.cpp]: This line is never reached, maybe we can remove it.

**Comment by groud** [comment]
This looks fine but I'm not sure it solves the problem. The issue was for nodes that are in the tree so `!is_inside_tree` is false.

**Comment by endragor** [comment]
It solved the problem in my game and in a sample project I made. Anchors are set within PackedScene::instance(), so the node is not yet inside tree.

**Comment by groud** [comment]
Yeah, but if you don't want margins to be modified when doing that you should use the p_keep_margin parameter and set it to true. The problem in #15635 is not when you set it through code

**Comment by endragor** [comment]
I don't do anything - just make a control with anchors (0,0,1,1) and zero margins and run it. The runtime does PackedScene::instance() when loading the scene.

**Comment by groud** [comment]
Ok then. But to be honest I think the problem comes from elsewhere, and this is not a proper fix. 
The margins should not be modified when the anchors are set, when saving a scene.

I this i should make p_keep_margin to true by default so that people don't get surprised.

**Comment by endragor** [comment]
This is NOT about saving a scene, it seems you still misinterpret something. It's about instancing a scene. When instancing a scene, the setters for anchors are called (because they are specified as node properties in the scene file), which result in margins being wrongly set because of incorrect _get_parent_range value. When loading a scene, setters are called with default parameters, so `p_keep_margin` is `false`.

**Comment by endragor** [comment]
`p_keep_margin` as `true` by default makes more sense to me from scripting perspective. When a control is edited in editor it's one thing, but in code it's not obvious that setting anchors will modify margins.

**Comment by groud** [comment]
Well, yeah saving or loading it doesn't not matter... The problem is this: 

> When loading a scene, setters are called with default parameters, so p_keep_margin is false.

So yeah, probably that p_keep_margin shoud be set to true by default.

**Comment by akien-mga** [comment]
Merging as it fixes a bug and I'm about to redo 3.0-rc1 binaries, but feel free to continue investigating the issue further.

**Pull request merged**